homeworks/17/instagram-api/Post.java

```diff
@@ -0,0 +1,173 @@
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+	 */
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		// TODO Auto-generated method stub
+        response.setContentType("text/html");
+        
+        try {
+	        String query = "SELECT * FROM posts WHERE user_id = ?";
+	        PreparedStatement statement = connection.prepareStatement(query);
+	        statement.setString(1, String.valueOf(user_id));
+		} catch (SQLException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+        String pathInfo = request.getPathInfo();
+        String[] splittedPath = pathInfo.split("/");
+        post_id = splittedPath[1];
+
+        if (splittedPath[2] == "comments") {
+            
+            try {
+                String query = "SELECT * FROM comments WHERE post_id = ?";
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setString(1, post_id);
+			} catch (SQLException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}           
+        }
+	}
+
+	/**
+	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+	 */
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		// TODO Auto-generated method stub
+		response.setContentType("text/html");
+
+        user_id = request.getSession(false).getAttribute("user_id").toString();
+        String image = request.getParameter("image");
+        String description = request.getParameter("description");
+
+        try {
+            String query = "insert into post(user_id, image, description)" + " values (?, ?, ?)";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user_id);
+            statement.setString(2, image);
+            statement.setString(3, description);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        String pathInfo = request.getPathInfo();
+        String[] split = pathInfo.split("/");
+        post_id = split[1];
+
+        if (split.length > 2) {
+            if (split[2] == "comments") {
+                if (split.length > 3) {
+                    comment_id = split[3];
+                }
+                try {
+                    String query = "INSERT INTO comments(post_id, description)" + " values (?, ?)";
+                    PreparedStatement statement = connection.prepareStatement(query);
+                    statement.setString(1, post_id);
+                    statement.setString(2, description);
+                    statement.executeUpdate();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
```

Comment:
There is a lot of stuff wrong here - query parameters are used, "if" cases are missing, path pattern matching won't work, etc. :disappointed: